### PR TITLE
added handling for the like case, so the Note in the list view is ref…

### LINKED ISSUE #4818: Notifications: Liked comment sometimes is not reflected in the detail view
### Expected behavior
When you tap like on a comment, the comment is liked and when you return to the notifications list and come back in, the liked status should be reflected.

### Actual behavior
The liked status is missed sometimes.

### Steps to reproduce the behavior
1. go to notifications tab
2. choose a comment
3. tap on it
4. like the comment
5. go back to the list
6. tap on the same comment
7. see the 'like' option is greyed, as if it hasn't been liked.

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4818

The like action was only being sent to the server and reflected on the detail view UI, but when coming back to the notifications list then tapping again, the object in the list adapter would still remain with the old data (i.e. unliked or liked, correspondingly).

So, what this PR does is:
- user likes a comment and the endpoint is hit
- when the call to the server returns successfully, we issue an EventBus signal with this information (which note has been liked)
- in the Notifications list screen, the signal is received and the note is re-fetched from the server and saved to the local DB.
- once in the DB, we get that note and update the list adapter with the newly fetched note.

This way, when the user taps on that note again, the right Like status will be shown.

**NOTE:** I tried an approach to save the note immediately to DB after hitting the like endpoint in the detail view (which would naturally be the simplest way), but that would mean re-building the note locally when it is safer to get the note build by the server, also following the general architecture for moderation (approve/unapprove/spam/trash) as it is in the app as of now.

To test:
1. go to notifications tab
2. choose a comment
3. tap on it
4. like the comment
5. go back to the list
6. tap on the same comment
7. see the 'like' option should be the way you left it on step 4 (liked or unliked respectively)

cc @daniloercoli 
also cc'ing @kwonye as this was first observed while tackling this issue #4782 ([see this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/4782#issuecomment-262546706))

### CODE REVIEW

**Comment by daniloercoli** [comment]
Just noticed that the base branch for this PR isn't develop. Shouldn't it be?

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java [hunk 1]
```diff
@@ -567,6 +569,27 @@ public void onErrorResponse(VolleyError error) {
...
     @SuppressWarnings("unused")
+    public void onEventMainThread(final NotificationEvents.NoteLikeStatusChanged event) {
+        // Like/unlike done -> refresh the note and update db
+        NotificationsActions.downloadNoteAndUpdateDB(event.noteId,
+                new RestRequest.Listener() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        EventBus.getDefault().removeStickyEvent(NotificationEvents.NoteLikeStatusChanged.class);
+                        //now re-set the object in our list adapter with the note saved in the updated DB
+                        Note note = NotificationsTable.getNoteById(event.noteId);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java]: It's unlikely that the note is null here, but better to have a check before replacing the note in the adapter.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java [hunk 2]
```diff
@@ -168,6 +168,11 @@ private Note getNoteAtPosition(int position) {
         return null;
     }
 
+    public void replaceNote(Note newNote) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java]: Might be better to have a check for `not null` as first thing in this method.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java [hunk 3]
```diff
@@ -168,6 +168,11 @@ private Note getNoteAtPosition(int position) {
...
+        int position = getPositionForNoteUnfiltered(newNote.getId());
+        mNotes.set(position, newNote);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java]: We should check that `position  != recyclerView.NO_POSITION` and just to be super sure `position <  mNotes.size()`

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java (hunk 1 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java]: addressed in 2612f76

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java (hunk 2 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java]: addressed in a2e6b04

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java (hunk 3 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java]: addressed in cd316c9

**Comment by mzorz** [comment]
Ready for another round @daniloercoli 🙇 thanks in advance :)

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**